### DISCUSSION TITLE
enhance requirements file parser or switch to another one

### DISCUSSION THREAD

**Discussion by jkowalleck** [CycloneDX/cyclonedx-python#319] [Ideas]
there are multiple discussions for alternative or missing features of the requirements file parser.

* https://github.com/CycloneDX/cyclonedx-python/discussions/273
* https://github.com/CycloneDX/cyclonedx-python/issues/318
* https://github.com/CycloneDX/cyclonedx-python/issues/315
* https://github.com/CycloneDX/cyclonedx-python/issues/284
* https://github.com/CycloneDX/cyclonedx-python/issues/194 & https://github.com/CycloneDX/cyclonedx-python-lib/issues/8

----

the story so far: the original requirements parser had the scope to understand the output of `pip freeze`.
but the people want more :-) 
* support for `-r` to chain multiple files
* support for custom package registries and repositories
* support for packages by relative path
* support for package hashes

there are suggestions to switch to an alternative requirements file parser at some point, as it supports some of the mentioned features.

----

lets discuss the topic here!

**Comment by llamahunter** [comment]
The docs for cyclonedx-bom says that it can read a "requirements.txt" file. If that's not the case, probably should be made clear.

But, disallowing private pypi repos seems pretty limiting.

**Reply by mostafa** [reply]
@llamahunter That's what I am trying to fix. For now, I'll focus on #315 and #284, but will eventually address all the issues.

**Reply by llamahunter** [reply]
See also #318. Don't care so much about local packages, but private repositories are critical for us since we have private libraries that we don't want to publish publicly just to use with cyclone-dx.

**Comment by jkowalleck** [comment]
fixed in v4
